packages/@aws-cdk/aws-s3-deployment/lib/lambda/index.py

```diff
@@ -115,8 +114,13 @@ def cfn_error(message=None):
 #---------------------------------------------------------------------------------------------------
 # populate all files from s3_source_zips to a destination bucket
 def s3_deploy(s3_source_zips, s3_dest, user_metadata, system_metadata, prune):
-    # create a temporary working directory
-    workdir=tempfile.mkdtemp()
+    # create a temporary working directory in /tmp or if enabled an attached efs volume
+    if ENV_KEY_IS_EFS_ENABLED in os.environ:
+        workdir = os.getenv(ENV_KEY_MOUNT_PATH) + "/" + str(uuid4())
+        os.mkdir(workdir)
+    else:
+        workdir = tempfile.mkdtemp()
+
```

Comment:
```suggestion
workdir = os.getenv('MOUNT_PATH', tempfile.mkdtemp())
```

Reply:
Done.

Reply:
We need to append uuid after `/mnt/lambda` so that two invocations extracts to different locations. With above approach i found permission conflict between two lambda invocations with efs both using `/mnt/lambda/contents` for extracting stuff.